src/App.jsx

```diff
@@ -1,14 +1,36 @@
+import { useState } from 'react';
+import ChatLog from './components/ChatLog';
+import initialMessages from './data/messages';
 import './App.css';
 
 const App = () => {
+  const senders = Array.from(new Set(initialMessages.map(initialMessage => initialMessage.sender)));
+  const chatHeader = `Chat between ${senders.join(' and ')}`;
+
+  const [messages, setMessages] = useState(initialMessages);
+
+  const toggleLike = (id) => {
+    setMessages((prevMessages) =>
+      prevMessages.map((message) =>
+        message.id === id
+          ? { ...message, liked: !message.liked }
+          : message
+      )
+    );
+  };
```

Comment:
Nice work on this function! By passing this down to your individual chat messages you are now able to have a single source of truth!